airbyte-integrations/bases/base-java/src/main/java/io/airbyte/integrations/destination/buffered_stream_consumer/BufferedStreamConsumer.java

```diff
@@ -141,11 +142,15 @@ protected void acceptTracked(final AirbyteMessage message) throws Exception {
         return;
       }
 
-      buffer.add(message);
-
-      if (buffer.size() == queueBatchSize) {
+      long messageSizeInBytes = Jsons.serialize(recordMessage.getData()).length() * 2; // Strings serialize to UTF-8 by default
```

Comment:
Maybe add a TODO here about optimizing the calculation of message byte size in the future.